File: src/Utiles/Update.ts
import {Player, Required} from '../types/MatchType';

export function overCompleted(players: Player[]) {
  var tempPlayers = players;
  var temp = players[0];
  tempPlayers[0] = players[1];
  tempPlayers[1] = temp;
  return tempPlayers;
}

/*update the required score and balls*/
export function updateRequiredScore(required: Required, runsScored: number) {
  var tempReq = required;
  let prevReqScore = required.numberOfRunsRequired;
  tempReq.numberOfRunsRequired = prevReqScore - runsScored;
  return tempReq;
}

/*update the score of the individual player and balls played*/
export function updatePlayerScore(
  players: Player[],
  playerPosition: number,
  runsScored: number,
) {
  var tempPlayers = players;
  var playertobeUpdated = tempPlayers[playerPosition];
  let prevPlayerScore = playertobeUpdated.numberOfRunsScored;
  let prevPlayedBalls = playertobeUpdated.numberofBallsPlayed;
  playertobeUpdated.numberOfRunsScored = prevPlayerScore + runsScored;
  playertobeUpdated.numberofBallsPlayed = prevPlayedBalls + 1;
  tempPlayers[playerPosition] = playertobeUpdated;
  return tempPlayers;
}

//update wickets in required
export function updateWickets(required: Required) {
  var temprequired = required;
  let prevWickets = temprequired.wicketsLeft;
  temprequired.wicketsLeft = prevWickets - 1;
  return temprequired;
}

File: android/app/src/main/java/com/testcricketmatch/Methods/Update.java
package com.testcricketmatch.Methods;

/**
 * Created by Harminder Singh on 07,October,2022
 */


import com.testcricketmatch.beans.Player;
import com.testcricketmatch.beans.Required;

import java.util.List;

public class Update {


    public static List<Player> overCompleted(List<Player> players){
        Player temp = players.get(0);
        players.set(0,players.get(1));
        players.set(1,temp);
        return players;
    }

    /*update the required score and balls*/
    public static Required updateRequiredScore(Required required, int runsScored){
        int prevReqScore = required.getNumberOfRunsRequired();
        required.setNumberOfRunsRequired(prevReqScore-runsScored);

        return required;
    }

    /*update the score of the individual player and balls played*/
    public static List<Player> updatePlayerScore(List<Player> players,int playerPosition,int runsScored){
        Player playertobeUpdated = players.get(playerPosition);
        int prevPlayerScore = playertobeUpdated.getNumberOfRunsScored();
        int prevPlayedBalls = playertobeUpdated.getNumberofBallsPlayed();
        playertobeUpdated.setNumberofBallsPlayed(prevPlayedBalls+1);
        playertobeUpdated.setNumberOfRunsScored(prevPlayerScore+runsScored);
        players.set(playerPosition,playertobeUpdated);
        return players;
    }

    //update wickets in required
    public static Required updateWickets(Required required){
        int prevWickets = required.getWicketsLeft();
        required.setWicketsLeft(prevWickets-1);
        return required;
    }

}
